packages/dapp/components/redeem/ucr-nft-redeem.tsx

```diff
@@ -67,12 +67,14 @@ const UcrNftRedeem = () => {
   };
 
   const redeemUcrNftForUad = async (amount: BigNumber) => {
-    const { debtCouponManager } = deployedContracts;
+    const contracts = await protocolContracts;
     const debtId = debtIds[selectedDebtId];
-    if (debtId && (await ensureERC1155Allowance("uCR-NFT -> DebtCouponManager", managedContracts.creditNft, signer, debtCouponManager.address))) {
-      await (await debtCouponManager.connect(signer).redeemCreditNft(debtId, amount)).wait();
-      refreshBalances();
-      fetchDebts(walletAddress, managedContracts.creditNft);
+    if (contracts.creditNft && contracts.creditNftManagerFacet) {
```

Comment:
can you achieve this in other way?

Reply:
> can you achieve this in other way?

`!` is banned and `?` also can't be used in that case